### ISSUE TITLE
SILE vs. TeX hyphenation patterns

### ISSUE THREAD

**Issue by Omikhleia** [sile-typesetter/sile#2104]
Things not addressed in #2102:

- Things where we have our own upgrades, so likely won't change?
   - Turkish (**tr**) : many changes by @alerque 
   - Esperanto (**eo**): solution by @ctrlcctrlv differing from what TeX currently has
- Things that might have to wait for a better implementation of BCP47...
   - en-GB / en-US (but see also just below, last item)
   - zh-Latn-pinyin
   - mul-ethi (Ethiopic)
   - nb/nn variants (Norsk, we do have them though a refactor could be neat eventually)
   - several la variants (liturgic/classic Latin)
   - mn (Mongolian in Cyrl script or Cyrl-x-lmc for Xalx Mongolian variant)
   - sh (Cyrl vs. Latn, Serbo-Croatian, now deprecated?)
- Things I am not sure of...
  - our "el" vs. TeX's "grc" is problematic
- Things where we differ but may change for better
  - pt (Portuguese): We have @jodros 's updates, but an incoming  PR might improve upon it: https://github.com/hyphenation/tex-hyphen/pull/62 **Should we wait or go for an early adoption?** = **0.16.x** or earlier ?
  - es (Spanish): Lots of differences with our patterns = **0.16.x** ?
  - th (Thai): Lots of differences with our patterns, and no idea where the latter come from. = (?)
  - bg (Bulgarian) = Lots of differences with our patterns, and no idea where the latter come from. = **0.16.x** ? Note the TeX patterns were updated in 2017 so whatever we had might be outdated (?)
  - de (German) = We have something, TeX has several variants depending on orthography reforms, frankly I don't know.
  - en (English, see also above): Currently we are based on en-US, but we lack a whole "additional patterns" that were not present in the original TeX file and were added later (though at an earlier stage than our import). My understanding is that "old" TeX being memory constrained, these were absent from the original implementation. But it's pretty unclear why we don't have the extra patterns... = **0.16.x** ? as it could make rendered documents different...

_Originally posted by @Omikhleia in https://github.com/sile-typesetter/sile/issues/2102#issuecomment-2336669874_

**Comment by jodros** [comment]
> pt (Portuguese): We have @jodros 's updates, but an incoming PR might improve upon it: https://github.com/hyphenation/tex-hyphen/pull/62 Should we wait or go for an early adoption? = 0.16.x or earlier ?

Glad to know about @leolca's work! I can't see reasons to wait.

**Comment by Omikhleia** [comment]
> Glad to know about @leolca's work! I can't see reasons to wait.

Yep, it really made my day when I read their (proposed) TUGboart article. It's scholarly, plenty of attempts, plenty of tables with number;.. It"s very interesting (even the things that they tried, judged inefficient, etc.)

**Comment by RJS32** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by Omikhleia** [comment]
We addressed some of the above-mentioned items in #2259 

What remains of this issue:
  - Handling BCP47 variants:
      - en-GB / en-US (but see also below)
      - zh-Latn-pinyin
      - mul-ethi (Ethiopic)
      - several la variants (liturgic/classic Latin)
      - mn (Mongolian in Cyrl script or Cyrl-x-lmc for Xalx Mongolian variant)
      - sh (Cyrl vs. Latn, Serbo-Croatian, now deprecated?)
- Somewhat but not totally dependent on BCP47
      - el > grc (indepdently tracked as #2123) + relatedboustrophedon package check
      - en-US (our current "en") not having the extended patterns.
      - de (several variants, but we are not up-to-date either)

**Comment by Omikhleia** [comment]
> What remains of this issue: (...)

A few 2026 updates regarding other things to track, in addition to the above:
 - kk (Kazakh in Cyrl script) now has supported [TeX hyphenation patterns](https://github.com/hyphenation/tex-hyphen/blob/master/hyph-utf8/tex/generic/hyph-utf8/patterns/tex/hyph-kk.tex) (added in 2026) = see #2375
 - Note that various existing patterns were also updated in 2025-2026
 - There's a good point regarding Finnish reported on [Typst's hypher](https://github.com/typst/hypher/pull/32), which could also apply to SILE (also currently using the "old" Saarinen patterns).
